Feature/22 temporary file

BaranovMykola opened the pull request:
Implemented feature #22

BaranovMykola committed:
Added creating temporary file in tmp directory. Stored cursor position,
path to temp file in kuzya settings.ini. Added reading and restoring
closed file, cursor position.

BaranovMykola committed:
Merged updated settings form

BaranovMykola committed:
Added checkbox "Reopen file on kuzya start" to settings. Added
conditions to saving and opening temp file. If "Reopen file on kuzya
start" is disabled Kuzya will work in old way

BaranovMykola committed:
Fixed bug with scaling optionsdialog form after added new element

BaranovMykola committed:
Added extenstion to temporary file using
Compiler::getSupportedExtenstions(Qstring lang). Removed qDebug(), add
opening temporary file instend of copy itself's content to textEditor.

BaranovMykola committed:
Removed some extra gaps in code

alexchmykhalo commented on src/optionsdialog.h:
```diff
@@ -99,6 +102,7 @@ private slots:
     QStringList localizationLanguageFilters;
 public:	
     bool isLineMarginVisible;
+    bool isReopenFile;
```
Please, don't do any public members in classes. Make setter\getter pair.

alexchmykhalo commented on src/optionsdialog.cpp:
```diff
@@ -549,6 +553,53 @@ QString OptionsDialog::readCompilerOptions(QString lang, QString comp)
     return options;
 }
 
+void OptionsDialog::writeTemporaryFileState()
```
Please, move writeTemporaryFileState() and readTemporaryFileState() out of OptionsDialog class.

We should stick to single responsibility principle (see SOLID for reference). Class called OptionsDialog - that means that it is dialog that contains UI for Kuzya options. That is its only responsibility. It should not take care of loading temporary files, dealing with main window configuration, etc.  This class should serve only one purpose - manage UI for setting Kuzya options.

BaranovMykola committed:
Moved writeTemporaryFileState() and readTemporaryFileState() methods to
Kuzya class in order to avoid OptionsDialog from extra duties. Moved
isReopenFile variable to private in OptionDialog. Added getter for this
variable.

BaranovMykola commented:
I've moved writeTemporaryFileState() and readTemporaryFileState() methods to
Kuzya class in order to avoid OptionsDialog from extra duties. Also i've moved
isReopenFile variable to private in OptionDialog. Added getter for this
variable.

BaranovMykola committed:
Removed declarations of some methods (wich was moved from
OptionsDialog). Added clearing of file just before writing in order to
protect of merging two parts of file (new and old) - file will be
rewrited definatly
